src/ansys/pyensight/core/listobj.py

```diff
@@ -141,13 +141,20 @@ def set_attr(self, attr: Any, value: Any) -> int:
 
         """
         count: int = 0
+        objid_list = []
+        session = None
         for item in self:
             if isinstance(item, ENSOBJ):
                 try:
-                    item.setattr(attr, value)
+                    if not session:
+                        session = item._session
+                    objid_list.append(item.__OBJID__)
```

Comment:
This works, but I feel that this for item loop would be faster if you did it like:

objid_list = [x.__OBJID__ for x in self if isinstance(x, ENSOBJ)]
count = len(objid_list)

for i in objid_list:
    if not session:
         session = item._session
    else:
        break
        
        
I don't know if large item lists are a concern or not for you, so up to you
@mariostieriansys

Reply:
@margalva thanks for the suggestion, you're right with the list comprehension I can make it faster